python/lsst/verify/compatibility/metricTask.py

```diff
@@ -0,0 +1,196 @@
+# This file is part of verify.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import abc
+
+import lsst.pex.config
+import lsst.pipe.base as pipeBase
+
+__all__ = ["MetricTask"]
+
+
+class MetricTask(pipeBase.Task, metaclass=abc.ABCMeta):
+    """A ``MetricTask`` object computes exactly one metric, based on
+    arbitrary input datasets.
+
+    Both the metric and the input data are configurable. Metrics may be
+    associated with a data ID at any level of granularity, including
+    repository-wide. MetricTasks should assume they take multiple input
+    datasets, regardless of the expected metric granularity, unless it is
+    impossible for more than one copy of a dataset to exist.
+
+    All input data shall be considered optional in order to maximize the
+    metric's usefulness for incomplete pipeline runs or runs with optional
```

Comment:
Honestly, both (i.e., one should have the freedom to request a metric without having to worry about whether its inapplicability will break things). But I can see why it would be confusing, and can adopt the latter wording.